### PR TITLE
frmw1305-suppressdepwarnings

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA: https://jira.rax.io/browse/FRMW-1305

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM

### CODE REVIEW

**Comment by CITguy** [comment]
@cathysiller: I'm still seeing deprecation warnings in the unit test output. I shouldn't see any deprecation warnings in unit tests.

File: src/utilities/rxEnvironment/scripts/rxEnvironment.spec.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ describe('service:rxEnvironment', function () {
     beforeEach(function () {
         // load module
         module('encore.ui.utilities');
+        module({ suppressDeprecationWarnings: true });
```
Comment (cathysiller) [inline src/utilities/rxEnvironment/scripts/rxEnvironment.spec.js]: instances of rxEnvironment.get() and rxEnvironmentUrl in rxEnvironment logic are causing deprecation warnings for deprecation unit tests -- added the true module to non-deprecated unit tests

File: src/utilities/rxPermission/scripts/rxPermission.spec.js [hunk 2]
```diff
@@ -16,6 +16,7 @@ describe('rxPermission', function () {
                     return $delegate;
                 });
             });
+            module({ suppressDeprecationWarnings: true });
```
Comment (cathysiller) [inline src/utilities/rxPermission/scripts/rxPermission.spec.js]: instances of Permission in rxPermission logic are causing deprecation warnings for deprecation unit tests -- added the true module to non-deprecated unit tests

File: src/utilities/rxUrlUtils/scripts/rxUrlUtils.spec.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ describe('service:rxUrlUtils', function () {
         module(function ($provide) {
             $provide.value('$route', route);
         });
+        module({ suppressDeprecationWarnings: true });
```
Comment (cathysiller) [inline src/utilities/rxUrlUtils/scripts/rxUrlUtils.spec.js]: instances of 'rxEnvironmentUrlFilter' in rxUrlUtils n logic are causing deprecation warnings for deprecation unit tests -- added the true module to non-deprecated unit tests

**Comment by flazz** [comment]
(Apologies being late to the game) why not have a warning service, and handle suppression local to the service?

File: src/components/rxOptionTable/scripts/rxOptionTable.js [hunk 4]
```diff
@@ -73,10 +73,12 @@ angular.module('encore.ui.rxOptionTable')
             emptyMessage: '@',
             disableFn: '&?'
         },
-        link: function (scope, element) {
```
Comment (flazz) [inline src/components/rxOptionTable/scripts/rxOptionTable.js]: cannot inject in link function, need to inject in directive itself.

File: src/utilities/routesCdnPath/scripts/routesCdnPath.js [hunk 5]
```diff
@@ -29,10 +29,12 @@ angular.module('encore.ui.utilities')
  * to use, but we are documenting it in case you find your own use case for it.
  *
  */
-.provider('routesCdnPath', function () {
-    console.warn(
-        'DEPRECATED: routesCdnPath will be removed in EncoreUI 4.0.0'
-    );
+.provider('routesCdnPath', function (suppressDeprecationWarningsProvider) {
```
Comment (CITguy) [inline src/utilities/routesCdnPath/scripts/routesCdnPath.js]: We should be able to inject directly into `$get()`

File: src/components/rxOptionTable/scripts/rxOptionTable.js (hunk 4 above)
Comment (cathysiller) [inline src/components/rxOptionTable/scripts/rxOptionTable.js]: updated

File: src/utilities/routesCdnPath/scripts/routesCdnPath.js [hunk 6]
```diff
@@ -30,13 +30,15 @@ angular.module('encore.ui.utilities')
...
-    );
-
     this.customURL = null;
 
-    this.$get = function () {
+    this.$get = function (suppressDeprecationWarnings) {
+        if (!suppressDeprecationWarnings) {
+            console.warn(
+                'DEPRECATED: routesCdnPath will be removed in EncoreUI 4.0.0'
+            );
```
Comment (CITguy) [inline src/utilities/routesCdnPath/scripts/routesCdnPath.js]: 👍

**Comment by parlarjb** [comment]
Vidyo LGTM

**Pull request merged**